### PR TITLE
fix: no negative currency cash register

### LINKED ISSUE #559: Cash Register Demo - floating-point error
**Describe the bug**
Demo of the Cash Register project has issue with the floating-point error. Causing it display values with error, and in certain situations, for remaining cash to go below `$0`.

**To Reproduce**
1. Go to https://cash-register.freecodecamp.rocks
2. Enter `330` and submit.
3. Enter `10` and submit.
4. Enter `4` and submit.
5. Enter `4` and submit.

At the step 3. value below status will include the number with clear floating-point error.
After step 5. in register _Nickels_ are below `$0`.

**Expected behavior**
Cash shouldn't go below `$0`, or be displayed with more than a dozen of digits after the dot.

**Screenshots**
image

**Additional context**
- The least time-absorbing solution is adding `toFixed(2)` method to two more places:
https://github.com/freeCodeCamp/demo-projects/blob/a712591fb5cd4cb07ec0efbd8dc2578f4893df1c/apps/cash-register/public/script.js#L66
https://github.com/freeCodeCamp/demo-projects/blob/a712591fb5cd4cb07ec0efbd8dc2578f4893df1c/apps/cash-register/public/script.js#L70

- More robust option is saying _no_ to floats and changing more internals to not do calculations on floats.

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #559

### CODE REVIEW

File: apps/cash-register/public/script.js [hunk 1]
```diff
@@ -62,11 +62,12 @@ const checkCashRegister = () => {
...
       let total = reversedCid[i][1];
-      while (total > 0 && changeDue >= denominations[i]) {
+      while (total - denominations[i] >= 0 && changeDue >= denominations[i]) {
         total -= denominations[i];
         changeDue = parseFloat((changeDue -= denominations[i]).toFixed(2));
         count++;
       }
+      
       if (count > 0) {
         result.change.push([reversedCid[i][0], count * denominations[i]]);
```
Comment (gikf) [inline apps/cash-register/public/script.js:72]:
```suggestion
        result.change.push([reversedCid[i][0], (count * denominations[i]).toFixed(2)]);
```
Rounding for status didn't survive merge conflict :slightly_smiling_face:

**Comment by a2937** [comment]
I linted the file so this can finally be merged.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Unfortunately this introduces a regression - the final test no longer passes for the demo project with these changes (there's likely an off-by-one error going on somewhere).

It may be beneficial to refactor the project to handle all of the math with cents as the base instead of dollars.

**Comment by gikf** [comment]
Hey @a2937

Thank you for opening this pull request.

As there was different PR - #602 - addressing linked issue merged recently, we are going to close this one.

Thank you and happy coding! 😊

---

If you have any questions, feel free to ask questions on the ['Contributors' category on our forum](https://forum.freecodecamp.org/c/contributors) or [the contributors chat room](https://discord.gg/PRyKn3Vbay).

**Pull request closed without merging**